src/imp/libc/io/io_slice.rs

```diff
@@ -1,85 +1,177 @@
-//! The following is derived from Rust's
-//! library/std/src/sys/unix/io.rs
-//! dca3f1b786efd27be3b325ed1e01e247aa589c3b.
-
-use super::super::c;
-use core::marker::PhantomData;
-use core::slice;
-
-#[derive(Copy, Clone)]
-#[repr(transparent)]
-pub struct IoSlice<'a> {
-    vec: c::iovec,
-    _p: PhantomData<&'a [u8]>,
-}
+pub use imp::*;
 
-impl<'a> IoSlice<'a> {
-    #[inline]
-    pub fn new(buf: &'a [u8]) -> IoSlice<'a> {
-        IoSlice {
-            vec: c::iovec {
-                iov_base: buf.as_ptr() as *mut u8 as *mut c::c_void,
-                iov_len: buf.len(),
-            },
-            _p: PhantomData,
-        }
+#[cfg(not(windows))]
+mod imp {
+    //! The following is derived from Rust's
+    //! library/std/src/sys/unix/io.rs
+    //! dca3f1b786efd27be3b325ed1e01e247aa589c3b.
```

Comment:
To keep this content easier to keep in sync with upstream, could you move the unix bits to a separate file, so that it isn't indented differently from upstream, and add the windows bits in their own file, with their own comment about which file in upstream Rust they're derived from?

Reply:
fixed